test/common/network/utility_test.cc

```diff
@@ -233,6 +244,47 @@ TEST(NetworkUtility, GetOriginalDst) {
   EXPECT_CALL(socket, addressType()).WillOnce(testing::Return(Address::Type::Pipe));
 #endif
   EXPECT_EQ(nullptr, Utility::getOriginalDst(socket));
+
+#ifdef SOL_IP
+  std::string addr;
+  sockaddr_storage storage;
+  auto& sin = reinterpret_cast<sockaddr_in&>(storage);
+  sin.sin_port = htons(9527);
+  testing::NiceMock<Api::MockOsSysCalls> os_sys_calls;
+  TestThreadsafeSingletonInjector<Api::OsSysCallsImpl> os_calls(&os_sys_calls);
+  ON_CALL(os_sys_calls, supportsIpTransparent()).WillByDefault(Return(true));
+  EXPECT_CALL(socket, addressType()).WillRepeatedly(Return(Address::Type::Ip));
+
+  sin.sin_family = AF_INET;
+  sin.sin_addr.s_addr = inet_addr("12.34.56.78");
+  socket.connection_info_provider_->setLocalAddress(
+      std::make_shared<Network::Address::Ipv4Instance>("1.2.3.4"));
+  EXPECT_CALL(socket, ipVersion()).WillRepeatedly(Return(Address::IpVersion::v4));
+  EXPECT_CALL(socket, getSocketOption(Eq(SOL_IP), Eq(SO_ORIGINAL_DST), _, _))
+      .WillOnce(DoAll(SetArg2Sockaddr(storage), Return(Api::SysCallIntResult{0, 0})))
+      .WillOnce(Return(Api::SysCallIntResult{-1, 0}));
+  EXPECT_CALL(socket, getSocketOption(Eq(SOL_IP), Eq(IP_TRANSPARENT), _, _))
+      .WillOnce(DoAll(SetArg2Int(1), Return(Api::SysCallIntResult{0, 0})));
```

Comment:
seems miss one more case for `getSocketOption` with `IP_TRANSPARENT` return failure.

Reply:
ok ,  will fix it tonight